src/comps/temp-withdraw.tsx

```diff
@@ -0,0 +1,240 @@
+import { DefaultNetworkConfigurations } from "@leather.io/models";
+import { useRouter } from "next/navigation";
+
+import { bytesToHex, hexToBytes } from "@stacks/common";
+
+import * as bitcoin from "bitcoinjs-lib";
+import { useAtomValue } from "jotai";
+import { bridgeConfigAtom, walletInfoAtom, WalletProvider } from "@/util/atoms";
+import {
+  bufferCV,
+  makeUnsignedContractCall,
+  PostConditionMode,
+  serializeTransaction,
+  tupleCV,
+  uintCV,
+} from "@stacks/transactions";
+import { serverBroadcastTx } from "@/actions/server-broadcast-tx";
+import {
+  callContractAsigna,
+  callContractFordefi,
+  callContractLeather,
+  callContractXverse,
+} from "@/util/wallet-utils";
+import { getStacksNetwork } from "@/util/get-stacks-network";
+
+const decodeBitcoinAddressToClarityRecipient = (
+  address: string,
+  network: DefaultNetworkConfigurations,
+) => {
+  const lower = address.toLowerCase();
+  const isMainnet = network === "mainnet";
+
+  // check if passed in address is bech32
+  let decodedBech32;
+  let isBech32 = false;
+  if (
+    lower.startsWith("bc1") ||
+    lower.startsWith("bcrt1") ||
+    lower.startsWith("tb1")
+  ) {
+    try {
+      decodedBech32 = bitcoin.address.fromBech32(address);
+      isBech32 = true;
+    } catch (err: any) {
+      throw new Error(`Invalid bech32 address: ${err.message}`);
+    }
+  }
+
+  if (isBech32 && decodedBech32) {
+    const {
+      prefix,
+      version: witnessVersion,
+      data: witnessProgram,
+    } = decodedBech32;
+
+    const hexHash = "0x" + bytesToHex(witnessProgram);
+    if (isMainnet) {
+      if (prefix !== "bc") {
+        throw new Error(`Expected mainnet bech32 (bc1...), got ${prefix}`);
+      }
+    } else if (!isMainnet) {
+      // regtest typically uses prefix 'bcrt'
+      if (prefix !== "bcrt") {
+        throw new Error(`Expected regtest bech32 (bcrt1...), got ${prefix}`);
+      }
+    }
+
+    if (witnessVersion === 0) {
+      if (witnessProgram.length === 20) {
+        // P2WPKH
+        return { version: "0x04", hashbytes: hexHash };
+      } else if (witnessProgram.length === 32) {
+        // P2WSH
+        return { version: "0x05", hashbytes: hexHash };
+      } else {
+        throw new Error(
+          `Unsupported witness v0 length=${witnessProgram.length}`,
+        );
+      }
+    } else if (witnessVersion === 1 && witnessProgram.length === 32) {
+      // P2TR
+      return { version: "0x06", hashbytes: hexHash };
+    } else {
+      throw new Error(
+        `Unrecognized witnessVersion=${witnessVersion}, length=${witnessProgram.length}`,
+      );
+    }
+  }
+};
+
+const data: NameKeysInfo[] = [
+  {
+    nameKey: "address",
+    type: "text",
+    initValue: "",
+    placeholder: "address to receive your Bitcoin",
+  },
+  {
+    nameKey: "amount",
+    type: "number",
+    initValue: "",
+    placeholder: "withdraw amount",
+  },
+  {
+    nameKey: "fee",
+    type: "number",
+    initValue: "",
```

Comment:
Can we have `0.0008` as default? Same as deposits, just to have a quicker flow

Reply:
consider it done